### PR TITLE
Add missing REAL/TINYINT/SMALLINT to JSON casts

### PR DESCRIPTION
**Pull request by idemura**
Issue #7981

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonOperators.java [hunk 1]
```diff
@@ -322,11 +325,40 @@ public static Slice castFromVarchar(@SqlType("varchar(x)") Slice value)
...
+    {
+        return internalCastFromLong(value, 12);
+    }
+
+    @ScalarOperator(CAST)
+    @SqlType(JSON)
     public static Slice castFromInteger(@SqlType(INTEGER) long value)
             throws IOException
     {
+        return internalCastFromLong(value, 20);
```
Comment (haozhun) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonOperators.java]: A more accurate estimate for tinyint/smallint/int seems to be 4/6/11. Do you think this is correct?

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestJsonFunctions.java [hunk 2]
```diff
@@ -317,4 +317,13 @@ public void testJsonEquality()
...
                 "JSON '{\"p_2\":\"v_2\", \"p_4\":true, \"p_1\": 1, \"p_3\":null, \"p_5\": {\"p_1\":1}}'", BOOLEAN, false);
     }
+    
+    @Test
+    public void testScalarToJson()
+    {
+        assertFunction("JSON '1.234' = CAST(1.234 AS JSON)", BOOLEAN, true);
+        assertFunction("JSON '1.234' = CAST(CAST(1.234 AS REAL) AS JSON)", BOOLEAN, true);
+        assertFunction("JSON '10' = CAST(CAST(10 AS TINYINT) AS JSON)", BOOLEAN, true);
+        assertFunction("JSON '10000' = CAST(CAST(10000 AS SMALLINT) AS JSON)", BOOLEAN, true);
```
Comment (haozhun) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestJsonFunctions.java]: Put these tests in `testCastFromIntegrals` or a new method that is called `testCastFromReal`

**Comment by wenleix** [comment]
Should we specify what are added in the commit log?

**Comment by haozhun** [comment]
@wenleix has a good point. The commit message should be more specific.

**Comment by idemura** [comment]
PTAL seems ok

**Comment by haozhun** [comment]
Change commit message to `Add cast from REAL/TINYINT/SMALLINT to JSON`

**Pull request merged**